### DISCUSSION TITLE
Limit rules to certain time during the day

### DISCUSSION THREAD

**Discussion by fberrez** [jertel/elastalert2#106] [Ideas]
The idea is to add a new attribute such as `limit_execution` to enable the alert to certain time during the week using a cron schedule expression.

```yaml
limit_execution: "'* 10-18 * * 1-5" # activates the rule from monday to friday, between 10am and 6pm.
```

See: https://github.com/Yelp/elastalert/issues/492

**Comment by nsano-rururu** [comment]
limit_execution was added in elastalert 0.2.0, is it a suggestion to change it?

**Reply by fberrez** [reply]
I can't find it in the [doc](https://elastalert2.readthedocs.io/en/latest/search.html?q=limit_execution&check_keywords=yes&area=default)

**Reply by nsano-rururu** [reply]
I don't know what happened in the past, but I guess I didn't write the documentation when I added the feature. so what?

**Reply by fberrez** [reply]
Shouldn't the documentation be updated?

**Reply by nsano-rururu** [reply]
@fberrez 

There is no rule who has to do it. Can you make a pull request to update the document separately?

**Reply by mrfroggg** [reply]
I think it was never merged in the master branch, only in a "beta" that never went out. Am I wrong?

**Reply by nsano-rururu** [reply]
@mrfroggg 

Beta has been merged into master.

https://github.com/Yelp/elastalert/issues/492#issuecomment-438024625
>I've merged this feature into a new branch, beta, and released it as a new package version 0.2.0b1 available on pypi.
>
>This includes a couple other changes as well, like threading support, but you can now limit rule execution to certain >times of the day using limit_execution using cron syntax. For example
>
>limit_execution: "* 7-22 * * *"
>Would mean to only run the rule between 7 am and 10 pm every day.
>
>This feature is still in beta, of course, but you're welcome to try.

Merged all beta changes into master
https://github.com/Yelp/elastalert/commit/14315e0989ba074f1979366517f27348efe2250f

Number of lines of limit_execution source code
https://github.com/jertel/elastalert2/blob/master/elastalert/elastalert.py#L1288
https://github.com/jertel/elastalert2/blob/master/elastalert/elastalert.py#L1358

**Reply by mrfroggg** [reply]
Right, I found it in the code!
This is something I'll try to test in the next days, I can try to do a quick documentation update on it!

**Comment by Fmstrat** [comment]
You have an extra quote in your sample.
